blivet/formats/fs.py

```diff
@@ -207,7 +207,13 @@ def free_space_estimate(cls, device_size):
             :return: estimated free size after format
             :rtype: :class:`~.size.Size`
         """
-        return device_size * cls._metadata_size_factor
+        estimate = device_size * cls._metadata_size_factor
+        try:
+            estimate = estimate * cls._compression_size_factor
+        except AttributeError:
+            # class doesn't have compression size factor, this is fine
+            pass
```

Comment:
If the attribute is missing the statement on 210 will fail as well. Also, is it fine to just swallow exceptions here?

Reply:
line 210 already exists, every class already defines a `_metadata_size_factor`. It just felt weird to define a `_compression_size_factor` of 1 for every class except btrfs. But actually, now I look at it, I think they all inherit from `FS`, so I can just set a factor of 1 there and we shouldn't need this.

Yes, swallowing exceptions is fine, since it just means we return the estimate without adjusting for compression. But anyway it's academic if I change it as described above.

Reply:
Got it, thanks for clarifying